httpcore5/src/main/java/org/apache/hc/core5/http/nio/ssl/TlsSupport.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.core5.http.nio.ssl;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+
+import org.apache.hc.core5.http.ssl.TLS;
+import org.apache.hc.core5.http.ssl.TlsCiphers;
+import org.apache.hc.core5.net.NamedEndpoint;
+import org.apache.hc.core5.reactor.ssl.SSLSessionInitializer;
+
+/**
+ * HTTP/1.1 TLS support methods
+ *
+ * @since 5.0
+ */
+public final class TlsSupport {
```

Comment:
I've never been a fan of classes named *Util, *Support, and so on. It just feel like a potential kitchen sink. How about a name focused what support is actually implemented? I do not have a specific name in mind.

Reply:
@garydgregory I do not like them either but with no specific names in mind I am not sure what we can do here.

Reply:
Hi @ok2c ,

Since the class operates on our SSLSessionInitializer in a focused manner, one alternative would be to use a (Java 8, I know) default method in that interface. I know Java 8 is a separate discussion but I really do not see what we gain by keeping support for a dead stale version of Java like 7.

The alternative, since the method returns a new

Reply:
@garydgregory We already have plenty of `Support` and `Utils` classes. There is already `H2TlsSupport`. Treating `TlsSupport` differently would just make it worse in my option.

Reply:
@ok2c OK, then all done.